docs/DeveloperGuide.md

```diff
@@ -1,34 +1,179 @@
+The professors are pressed for time and they require a simple software to organize their monthly events
+and keep track of their students' attendance and performance. 
+
+### Value proposition
+Our application will reduce the stress of the professor by allowing them to easily enter and store
+their work schedule as well as their students' records. After storing the data, the Professor can have
+quick access to the information in either a list or a calendar view. 
+
+### Audience 
+The design documentation is in general for anyone who wants to understand the system architecture and design of 
+PAC. The following groups are in particular the intended audience of the document.
+- PAC project managers
+- PAC developers
+- PAC software testers
+
+## Architecture
+This section presents the architecture of PAC. It explains the architecture of main components of PAC.
+
+### Overall Architecture
+*Overall Class diagram*
+{To be added in future revisions}
+
+### Command
+![Command](images/Command.png "Class diagram of Command component")         
+*Class diagram of the Command component*  
+
+Commands are the main classes to be executed in PAC. All of the specific Command classes inherit the 
+base Command abstract class, and utilize its abstract execute() method.  
+A subclass of Command is created and executed when the professor input a corresponding command.
+ 
+### Parser
+*Class diagram of the Parser component*  
+There are total of four Parser classes as shown below. Each Parser class correspond to a feature 
+of PAC. 
+
+| Parser                    | Created in                                                    |
+|---------------------------|---------------------------------------------------------------|
+| EventParser               | EventCommandInterpreter                                       | 
+| CalenderParser            | EventCommandInterpreter                                       | 
+| AttendanceParser          | Step-by-step command at Attendance-related command classes    |
+| PerformanceParser         | Step-by-step command at performance-related command classes   |  
+
+A Parser class is created when a user input contains data to be stored or used in certain features.    
 
-## Design & Implementation
+### UI
+![Ui](images/Ui.png "Class diagram of Ui component")                
+*Class diagram of the UI component*   
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+UI is the main class handles user display, which includes reading user input and printing information 
+back to the user on command-line.  
+Besides the normal command line messages, there are two subclasses of UI: 
+DisplayList and DisplayTable, to specifically print the list and table interface to professor. 
 
+## Feature Design and Implementation 
+### Event
+![event](images/event.png "Class diagram of Event component")           
+*Class diagram of the Event component*
```

Comment:
This diagram is informative and easier to follow. However, since this is auto-generated, there are some aspects that auto generated diagrams cannot express some terms in words like accessibility in methods (public, private, etc), and they refer to icons instead